src/centralserver/admin-service/core/src/test/java/org/niis/xroad/centralserver/restapi/service/ConfigurationSigningKeysServiceImplTest.java

```diff
@@ -35,49 +35,133 @@
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
-import org.niis.xroad.cs.admin.api.domain.ConfigurationSigningKey;
+import org.niis.xroad.centralserver.restapi.service.exception.NotFoundException;
+import org.niis.xroad.centralserver.restapi.service.exception.SigningKeyException;
+import org.niis.xroad.cs.admin.api.facade.SignerProxyFacade;
 import org.niis.xroad.cs.admin.core.entity.ConfigurationSigningKeyEntity;
 import org.niis.xroad.cs.admin.core.entity.ConfigurationSourceEntity;
 import org.niis.xroad.cs.admin.core.entity.mapper.ConfigurationSigningKeyMapper;
 import org.niis.xroad.cs.admin.core.entity.mapper.ConfigurationSigningKeyMapperImpl;
 import org.niis.xroad.cs.admin.core.repository.ConfigurationSigningKeyRepository;
+import org.niis.xroad.restapi.config.audit.AuditDataHelper;
+import org.niis.xroad.restapi.config.audit.AuditEventHelper;
+import org.niis.xroad.restapi.config.audit.RestApiAuditProperty;
 
+import java.time.Instant;
 import java.util.List;
 import java.util.Map;
+import java.util.Optional;
 
 import static ee.ria.xroad.signer.protocol.dto.TokenStatusInfo.OK;
-import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.mock;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditEvent.DELETE_EXTERNAL_CONFIGURATION_SIGNING_KEY;
+import static org.niis.xroad.restapi.config.audit.RestApiAuditEvent.DELETE_INTERNAL_CONFIGURATION_SIGNING_KEY;
 
 @ExtendWith(MockitoExtension.class)
 class ConfigurationSigningKeysServiceImplTest {
 
+    @Mock
+    private AuditEventHelper auditEventHelper;
     @Mock
     private ConfigurationSigningKeyRepository configurationSigningKeyRepository;
+    @Mock
+    private AuditDataHelper auditDataHelper;
+    @Mock
+    private SignerProxyFacade signerProxyFacade;
+
     private final ConfigurationSigningKeyMapper configurationSigningKeyMapper = new ConfigurationSigningKeyMapperImpl();
 
     private ConfigurationSigningKeysServiceImpl configurationSigningKeysServiceImpl;
 
     @BeforeEach
     public void setup() {
-        configurationSigningKeysServiceImpl =
-                new ConfigurationSigningKeysServiceImpl(configurationSigningKeyRepository, configurationSigningKeyMapper);
+        configurationSigningKeysServiceImpl = new ConfigurationSigningKeysServiceImpl(configurationSigningKeyRepository,
```

Comment:
Why recreate it before each test? configurationSigningKeysServiceImpl can be instantiated on class level.

Reply:
Did not use class level initialization here, because it only works with `@InjectMocks`, which in turn does not work with `configurationSigningKeyMapper` regular object.
Come to think of it, a workaround would be to annotate `configurationSigningKeyMapper` with `@Spy`